src/libraries/System.Net.Sockets/src/System/Net/Sockets/TCPListener.cs

```diff
@@ -274,7 +275,7 @@ private void CreateNewSocketIfNeeded()
                 _serverSocket.ExclusiveAddressUse = true;
             }
 
-            if (_allowNatTraversal != null)
+            if (_allowNatTraversal != null && OperatingSystem.IsWindows())
```

Comment:
Are we concerned about behavior change here at all? Previously someone would get a `PNSE` during socket creation, if they called `TCPListener.AllowNatTraversal(bool allowed)`. Now it looks like it will no longer PNSE.